### ISSUE TITLE
Normal Search results in infinite loop in sdk 1.4

### ISSUE THREAD

**Issue by sunieldalal** [splunk/splunk-sdk-java#77]
Running search something like this and results into infinite loop on !isDone() although original job has been completed on Splunk Server. Same code works fine with SDK 1.3.1

```
LOGGER.info("\n**************** Normal Search Starts ****************\n");
		Job job = _getService().getJobs().create(searchQuery, jobargs);
		LOGGER.info("Job Id is: " + job.getSid());
		
		// Wait for the search to finish
		while (!job.isDone()) {
		    try {
		    	LOGGER.info(" Still waiting for the job to complete ... ");
		        Thread.sleep(10000);
		    } catch (InterruptedException e) {
		        // TODO Auto-generated catch block
		        e.printStackTrace();
		    }
		}
```

**Comment by shakeelmohamed** [comment]
Hi @sunieldalal we will look into this. Thanks for letting us know

**Comment by shakeelmohamed** [comment]
Hi @sunieldalal,

Can you try adding `job.refresh();` before your `Thread.sleep(10000);`? This will actually update the `Job` objects status from Splunk.

Here's an example program that works for me

```java
public class Program {
    public static void main(String[] args) {
        Command command = Command.splunk("info").parse(args);
        Service service = Service.connect(command.opts);

        Job job = service.getJobs().create("search index=_internal");
        System.out.println(job.getSid());
        int i = 0;
        // Wait for the search to finish
        while (!job.isDone()) {
            try {
                System.out.println("Try: " + i++);
                job.refresh();
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                // TODO Auto-generated catch block
                e.printStackTrace();
            }
        }

        System.out.println("Number of events found: " + job.getEventCount());
    }
}
```

**Comment by sunieldalal** [comment]
Thanks a lot Shakeel for your time. I will have a look and will let you
know if I face any issues still. You can close the issue as of now.

**Comment by shakeelmohamed** [comment]
No problem, glad I can help :)

**Comment by shakeelmohamed** [comment]
@sunieldalal we've released v1.5.0 which addresses this issue.
